packages/accounts-password/password_tests.js

```diff
@@ -1392,6 +1392,75 @@ if (Meteor.isServer) (function () {
       }, /Incorrect password/);
     });
 
+  Tinytest.add(
+    'passwords - reset password should work when token is not expired',
+    function (test, onComplete) {
+      var username = Random.id();
+      var email = username + '-intercept@example.com';
+
+      var userId = Accounts.createUser({
+        username: username,
+        email: email,
+        password: "old-password"
+      });
+
+      var user = Meteor.users.findOne(userId);
+
+      Accounts.sendResetPasswordEmail(userId, email);
+
+      var resetPasswordEmailOptions =
+        Meteor.call("getInterceptedEmails", email)[0];
+
+      var re = new RegExp(Meteor.absoluteUrl() + "#/reset-password/(\\S*)");
+      var match = resetPasswordEmailOptions.text.match(re);
+      test.isTrue(match);
+      var resetPasswordToken = match[1];
+
+      makeTestConnection(
+        test,
+        function () {
+          
+          test.isTrue(Meteor.call("resetPassword", resetPasswordToken, "new-password"));
+          test.isTrue(Meteor.call("login", {user: {username: username}, password: "new-password"}));
+
+          onComplete();
```

Comment:
This test completes prematurely, without calling `onComplete()`, since you used `Tinytest.add` instead of `Tinytest.addAsync`. Unfortunately, that's not the only problem: `makeTestConnection` doesn't modify the default connection used by `Meteor.call`, so these method calls don't work as written.